### ISSUE TITLE
Global search with StPipe : error

### ISSUE THREAD

**Issue by poupougnac** [lorenzofox3/Smart-Table#562]
Hi,

When I use StPipe with StSearch (no argument ie global search) it throws this error : 
 
s(...).assign is not a function

It would be better to create an object like "globalSearch":value or something similar to pass to the server.

**Comment by lorenzofox3** [comment]
sorry I have no idea of what you are talking about, can you please provide an running example as explained in the "submit an issue" section

**Comment by poupougnac** [comment]
Apologies... I did find why and it's not related directly to smart-table.
I'm using jade with partials.
It compiles : 

input(st-search) 

in

<input st-search="st-search" /> 

because jade cannot detect the doctype for partials and try to avoids syntax error by defining a default value to the attribut. 

But I've found something interesting though... The error was thrown due to an attribut value containing a "-".

<input st-search="my-value"/>

will throw an error when you type some text.

http://plnkr.co/edit/7AtUfHyFSK2Mxr5LYXgb?p=preview
